proxy_http.go

```diff
@@ -154,9 +154,15 @@ func (proxy *proxy) handle(ctx context.Context, downstreamIn io.Reader, downstre
 		defer tr.CloseIdleConnections()
 		next = func(ctx filters.Context, modifiedReq *http.Request) (*http.Response, filters.Context, error) {
 			modifiedReq = modifiedReq.WithContext(ctx)
+			modifiedReq = prepareRequest(modifiedReq)
+
+			// Note that the following request aware handling only applies when the upstream
+			// connection has already been made -- i.e. when there is a net.Conn that is possibly
+			// request aware. In particular it does not apply on the first request because the
+			// RoundTrip call creates the upstream connection in that case. See DialContext above.
 			setRequestForAwareConn(ctx, modifiedReq)
 			handleRequestAware(ctx)
-			resp, err := tr.RoundTrip(prepareRequest(modifiedReq))
+			resp, err := tr.RoundTrip(modifiedReq)
```

Comment:
The above fixes an issue with multiple requests on the same connection where previously all Proxy-Connection changes after the first request would get clobbered by prepareRequest killing the Connection header regardless. 

This also makes request aware handling consistent for the first request and subsequent requests. Previously on the first request prepareRequest would be called before anything request aware, while on subsequent requests it was reversed. Now prepareRequest is always called first.

Reply:
Worth a unit test?

Reply:
yeah will do!